### ISSUE TITLE
Firestore in not working on Android 4.3.

### ISSUE THREAD

**Issue by umarpazir11** [firebase/quickstart-android#515]
Describe your environment

  * Android device: Urovo Smart POS Payment Terminal i9000S
  * Android OS version: 4.3
  * Google Play Services version: 3.2.0'
  * Firebase/Play Services SDK version: 'com.google.firebase:firebase-firestore:17.0.1'
  
Describe the problem:
I am trying to insert data on android 4.3 device and its not working but its working on API level 21 and greater. And there is no error in logcat.

#### Steps to reproduce:

  1. Run firestore code to insert data in android 4.3 device. 

#### Observed Results:
No error in logcat.
  
  
#### Expected Results:
  * What did you expect to happen?
    Data insertion in firestore.
#### Relevant Code:
 ```
FirebaseFirestore db = FirebaseFirestore.getInstance();
  db.collection("TransPayment:" +prefManager.getDeviceID()).document(transPayment.getTransactionId())
                .set(transPaymentList.get(indexTrans))
                .addOnSuccessListener(documentReference ->{
                    Log.d("Firestore", "DocumentSnapshot added with ID: ");
                })
                .addOnFailureListener(e -> {
                    Toast.makeText(LogActivity.this, "Firestore Trans Error", Toast.LENGTH_SHORT).show();
                    Log.w("Firestore", "Error adding document", e);
                });

**Comment by samtstern** [comment]
@umarpazir11 does this work on another device but not on the Android 4.3 device?

**Comment by samtstern** [comment]
@umarpazir11 also could you try enabling debug logging for Firestore and then showing the logcat?
https://firebase.google.com/docs/reference/android/com/google/firebase/firestore/FirebaseFirestore.html#setLoggingEnabled(boolean)

**Comment by samtstern** [comment]
@umarpazir11 closing this issue since it's been 27 days since last response.  If you want to re-open it just comment and let me know.

**Comment by rmushfiqur2** [comment]
Hello, I am having same issure. Firestore is working on lollipop + but in kitkat it doesn't work

**Comment by samtstern** [comment]
@rmushfiqur2 can you consistently reproduce the issue in a simple project?  If so could you provide steps to reproduce?  We could re-open this issue.

**Comment by samtstern** [comment]
In order to debug this issue further we need debug logs from Firestore running on a 4.3 device where it is not working.

**Comment by JassielMelgoza** [comment]
This is wrong: "TransPayment:" +prefManager.getDeviceID()
Use this: "TransPayment/" +prefManager.getDeviceID()

**Comment by ericksprengel** [comment]
`In App Messaging` dependency was the root cause for my project.
I updated it from `17.0.0` to `17.1.1`:
```
-    implementation 'com.google.firebase:firebase-inappmessaging-display:17.0.0'
+    implementation 'com.google.firebase:firebase-inappmessaging-display:17.1.1'
```

### More information about this analysis:
I enabled logging for Firestore in Android: `FirebaseFirestore.setLoggingEnabled(true)`
Then I got the root cause in the logcat (see below crash stack).
Googling it, I found an issue in Firebase Android SDK related to `In App Messaging` with a similar crash stack trace: https://github.com/firebase/firebase-android-sdk/issues/244
Someone told that it was caused by a dependency conflict between Firestore and In App Messaging, and another one told it was fixed in version `17.1.1`:
```
              Firestore  W  (19.0.0) [OnlineStateTracker]: Could not reach Cloud Firestore backend. Connection failed 1 times. Most recent error: Status{code=UNAVAILABLE, description=null, cause=javax.net
                            .ssl.SSLHandshakeException: javax.net.ssl.SSLProtocolException: SSL handshake aborted: ssl=0x5332cd50: Failure in SSL library, usually a protocol error
                         W  error:140740B5:SSL routines:SSL23_CLIENT_HELLO:no ciphers available (external/openssl/ssl/s23_clnt.c:486 0x4171476a:0x00000000)
                         W      at org.apache.harmony.xnet.provider.jsse.OpenSSLSocketImpl.startHandshake(OpenSSLSocketImpl.java:412)
                         W      at io.grpc.okhttp.OkHttpProtocolNegotiator.negotiate(OkHttpProtocolNegotiator.java:92)
                         W      at io.grpc.okhttp.OkHttpProtocolNegotiator$AndroidNegotiator.negotiate(OkHttpProtocolNegotiator.java:147)
                         W      at io.grpc.okhttp.OkHttpTlsUpgrader.upgrade(OkHttpTlsUpgrader.java:63)
                         W      at io.grpc.okhttp.OkHttpClientTransport$2.run(OkHttpClientTransport.java:474)
                         W      at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
                         W      at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1076)
                         W      at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:569)
                         W      at java.lang.Thread.run(Thread.java:856)
                         W  Caused by: javax.net.ssl.SSLProtocolException: SSL handshake aborted: ssl=0x5332cd50: Failure in SSL library, usually a protocol error
                         W  error:140740B5:SSL routines:SSL23_CLIENT_HELLO:no ciphers available (external/openssl/ssl/s23_clnt.c:486 0x4171476a:0x00000000)
                         W      at org.apache.harmony.xnet.provider.jsse.NativeCrypto.SSL_do_handshake(Native Method)
                         W      at org.apache.harmony.xnet.provider.jsse.OpenSSLSocketImpl.startHandshake(OpenSSLSocketImpl.java:371)
                         W      ... 8 more
                         W  }
                         W  This typically indicates that your device does not have a healthy Internet connection at the moment. The client will operate in offline mode until it is able to successfully co
                            nnect to the backend.
```
